Initial Documentation setup

jorgelainfiesta opened the pull request:
## Approach

**So far I**:
- Ran @docusaurus/init@next init template
- Moved Min's docs to the folder structure used by docusaurus
- Adjusted the sidebar with Min's current docs
- Removed the blog

**Pending**:
- [x] Custom styles
- [x] Deployment along-side API and homepage.

minkimcello committed:
Add pre-pre drafts for getting started and interactor guide

jorgelainfiesta committed:
Inits docusaurus

jorgelainfiesta committed:
Arranges Min's docs into Docusaurus

jorgelainfiesta committed:
Sets netlify build command

jorgelainfiesta committed:
Build with docusaurus on netlify's config

jorgelainfiesta committed:
Cleans up .gitignores for new website setup

jorgelainfiesta committed:
Sets up basic typography

jorgelainfiesta committed:
Makes link references / for a valid build

minkimcello committed:
Add development markdowns

Remove unnecessary docusaurus command
Delete mdx.md template page

minkimcello committed:
Fix empty link

minkimcello committed:
Create concept category and a few empty pages

minkimcello committed:
WIP Add more notes

jorgelainfiesta commented on website/docs/docs/getting-started/1-installation.md:
````diff
@@ -0,0 +1,19 @@
+$ yarn add -D bigtest
+```
+
+Once that's finished installing, you can run the following command:
````
Where should I run this? is it the within the project that I want to test or is it arbitrary?

minkimcello replied:
Both commands can be run from anywhere in the project. I assume it does a thing where it searches up the directory until it hits package.json because you can run `yarn bigtest init` from `./test/` or even `./node_modules` and it'll create/modify `bigtest.json` at the root of the project.

I'll need to confirm how it works in a monorepo.

jorgelainfiesta commented on website/docs/docs/getting-started/2-your-first-test.md:
````diff
@@ -0,0 +1,41 @@
+
+To demonstrate how you can write tests, let's install the TodoMVC app from BigTest:
+```
+$ yarn add -D @bigtest/todomvc
````
When we install this, is it to the project initiated by BigTest?

minkimcello replied:
I don't think bigtest is initiating a project. I think the assumption is people would be adding bigtest _to_ their pre-existing project.

So that being said, we're adding the todomvc app as a dev dependency (presumably to their pre-existing app) just so they can run bigtest against it. It's for demonstration purposes only and people should remove/delete it once they're done following along and understand how it works.

All they would need to do is remove the dependency of todomvc and modify the start script in package json.

jorgelainfiesta commented on website/docs/docs/getting-started/1-installation.md:
````diff
@@ -0,0 +1,19 @@
+$ yarn bigtest init
+```
+
+This command will help generate the `bigtest.json` file for you.
````
Is generating the bigtest.json the only thing that happens with this command?

minkimcello replied:
Yeah. And the command will modify the bigtest.json file if it already exists. And if the bigtest.json file already exists, running bigtest init will make suggestions based on what your previous configurations are.

For example, when you run `bigtest init` for the first time (and there's no bigtest.json file), the default value for the app port will be 3000. But if you already have bigtest.json configured to run the app in 4000, running bigtest init will then go "Where would you like to run your application? 4000?"

jorgelainfiesta commented on website/docs/docs/getting-started/2-your-first-test.md:
````diff
@@ -0,0 +1,41 @@
+
+Assuming you went with the default settings for `bigtest.json`, the app command should be configured as `yarn start`:
+```json
+{
````
It this json the default content from `bigtest.json`?

minkimcello replied:
No, I omitted some of the properties of the full default config file. If you run `bigtest init` and hit enter all the way through (so it's only default suggested values), this is what it would look like:
```json
{
  "port": 24000,
  "launch": [
    "chrome.headless"
  ],
  "app": {
    "command": "yarn start",
    "env": {
      "PORT": 3000
    },
    "url": "http://localhost:3000"
  },
  "testFiles": [
    "test/**/*.test.{ts,js}"
  ]
}
```

jorgelainfiesta commented on website/docs/docs/getting-started/2-your-first-test.md:
````diff
@@ -0,0 +1,41 @@
+  }
+}
+```
+Go into your `package.json` file and modify the start script as `yarn bigtest-todomvc 3000`:
````
I think I'm lost here, but perhaps because I don't know to whom this package.json belongs to. Might be clarified from the question above of where this is happening.

minkimcello replied:
Okay so to give a better idea of what I was trying to do, I made some diagrams:

Say I have `Min's Website` and I want to try out bigtest. Normally I would install bigtest and configure it to my website. The red arrow indicates configuration.
![Screen Shot 2020-11-09 at 9 24 25 AM](https://user-images.githubusercontent.com/29791650/98553518-b053e580-226d-11eb-9931-5066058376db.png)

The idea is we want to guide them step by step as much as possible but if we were to make a guide that works for any app/website, we have to be more vague. We can't help them write test for a headings at the top of the website because some websites for some reason might not have a heading. So that's why we're slapping on the todomvc app on `Min's Website` so people can practice writing test directly from their project.

Another strategy would be to just have readers clone the todomvc app and write tests for it from the todomvc repo directly. But I think people want to dive into it as soon as possible. Meaning when there's a tool and the guide tells you to `git clone ourTools-tutorial.git`, I end up skipping it and just reading the documentation. But if the tutorial lets you put it on your project as soon as possible, it's more encouraging. We can discuss that theory further.

> The summary of the last two paragraphs: put it in your actual repo asap = "okay let's do this!". make people clone an example repo for practice = "... eh maybe i'll do it some other time. too much of a hassle."

So then when people add the todomvc app to their repo, it looks like this:
![Screen Shot 2020-11-09 at 9 25 25 AM](https://user-images.githubusercontent.com/29791650/98554430-b26a7400-226e-11eb-9580-a3ed631795bd.png)

Instead of configuring bigtest to Min's Website, we're just configuring it to todomvc app. And once people go through the guides, we can instruct them to remove todomvc app and fix the start script in their package.json to react-scripts start or ng serve or whatever it is they're using.

![Screen Shot 2020-11-09 at 9 26 23 AM](https://user-images.githubusercontent.com/29791650/98554644-f493b580-226e-11eb-8da9-30e039a3daba.png)

jorgelainfiesta replied:
It seems a bit counterintuitive to install an entire app in your app to try out a framework, but the rationale you provide is very strong. Perhaps we can just make clear from the beginning that that's the approach that we'll take, and even add a small note on the fact that they can also clone the project if it makes them feel more comfortable. 

I can try to write that down so you can focus on new content, I'm asking to understand what you want to say before I bring it to the editor and such. With this context that you provided now I can write out some more details on the guide. Thanks!

jorgelainfiesta commented on website/docs/docs/getting-started/2-your-first-test.md:
````diff
@@ -0,0 +1,41 @@
+
+To demonstrate how you can write tests, let's install the TodoMVC app from BigTest:
+```
+$ yarn add -D @bigtest/todomvc
````
@cowboyd is there a way we can extract `todomvc` from the monorepo?  I think it'd be better if we renamed it `accessible-todomvc` for example, instead of linking it to bigtest directly. In my opinion, the current setup smells fishy because you're using a bigtest app to test bigtest. Makes me wonder if it'll run on any other app or not.

Plus, we hint a lot of stuff on the `accesible-todomvc` name: that we care a11y, that your app must follow good practices to use bigtest

jorgelainfiesta commented on website/docs/docs/getting-started/2-your-first-test.md:
````diff
@@ -0,0 +1,41 @@
+  .assertion(Heading('todos').exists());
+```
+
+That's it! Now you can run `yarn bigtest ci` and you should see one passing step and one passing assertion.
````
I'm a bit confused about the relationship between bigtest cli because we had to modify somebody's package json to make a `start`.

minkimcello replied:
Between bigtest cli and?

So there were two ways to run todomvc app instead of the reader's app: modify the start script of package.json or modify the app.command in bigtest.json. I chose to go with modifying package.json because we all know how that works.

Once they're done with the tutorial, it'll be easier for someone to realize that the start script needs to be re-adjusted for their own app versus going into bigtest.json and realize that the app.command in bigtest.json is connected to the start script in package.json.

If the guide tells them to change the app.command script to `yarn bigtest-todomvc 3000`, they might think it's a bigtest-related command that needs to stay in their configuration.

jorgelainfiesta replied:
Ok great info, thanks! clarifies a lot

jorgelainfiesta reviewed (commented):
Nice work so far! I left a few comments on the Getting Started section.

minkimcello committed:
Jen: Interactor docs getting-started

minkimcello committed:
Add tabs and old notes

minkimcello committed:
Add more content to getting started

minkimcello committed:
Add script and rename files for organizing purposes

minkimcello committed:
Add more notes, write content for built-in doms and locators, filters, actions section

minkimcello committed:
Bigtest alpha in tabs

jenweber committed:
Fill in more of the quick start

jenweber committed:
Wording improvements and comments cleanup

jorgelainfiesta committed:
Separates documentation for Interactors and Platform

minkimcello committed:
Change const declarations to let

minkimcello committed:
Add old notes and rearrange pages

minkimcello committed:
Reorganize again to slim down platform

minkimcello committed:
WIP Interactor docs

jenweber committed:
Add marketing and community language

jenweber committed:
Add integrations overview

jenweber committed:
Links to Jest and cypress

jenweber committed:
More details for write your own

jenweber committed:
Edits to locators-filters-actions

minkimcello committed:
Add comments for meeting

minkimcello committed:
Remove duplicate quick start and move quick start up one directory

minkimcello committed:
Remove why bigtest and move comments to interactor landing page

minkimcello committed:
Rename overview to introduction page

minkimcello committed:
Add complex interactor example of tablecell and add comments

minkimcello committed:
Update notes

minkimcello committed:
Fix broken link

minkimcello committed:
Update notes from response from list of questions

minkimcello committed:
Configure workflows to not run on this branch to save resources

minkimcello committed:
Add notes for Jen

jenweber committed:
Edits to locator narrative

jenweber committed:
Interactors landing page

jenweber committed:
interactors landing page pitch

minkimcello committed:
Rearrange pages and add more notes

minkimcello committed:
Fix build error and update notes

jenweber committed:
Fill in the Built in DOM article

jenweber committed:
Fill in integrations docs and rename page

jenweber committed:
Iterations on the locators-filters-actions

jenweber committed:
Fix Get Started link

cowboyd committed:
Add about page

minkimcello committed:
Fix broken links and add slug

minkimcello committed:
Disable workflows

minkimcello committed:
Add/modify contents for pages 1, 2, 3

minkimcello committed:
Update notes

minkimcello committed:
More content for writing your own interactor page

jenweber committed:
Fill in internal links

jenweber committed:
Draft Cypress integration section

jenweber committed:
Clean up code comments

minkimcello committed:
Fix disabling workflows with ! syntax

minkimcello committed:
Add sample app in quick start

minkimcello committed:
Fix link to url from file names

minkimcello committed:
Add asserting section in filters to cover has and is

minkimcello committed:
Small changes to write-your-own page

minkimcello committed:
Add more content to integration page and update with correct cypress links

minkimcello committed:
Update notes

minkimcello committed:
Fix broken links

minkimcello committed:
More changes to locators-filters-actions page

minkimcello committed:
Relocate bigtest platform shout out in quick start

minkimcello committed:
Add information on the page interactor in built-in page

minkimcello committed:
Add complex example in write-your-own

minkimcello committed:
Enhance platform docs

jbolda committed:
feat: typedoc to docusaurus

jorgelainfiesta committed:
Landing page styles for Interactors

jorgelainfiesta committed:
Navigation styles

jorgelainfiesta committed:
Adjusts mobile styles

jbolda committed:
scope back down to only interactors

It seems there are issues with monorepos using this plugin. We can't use it multiple times. We need interactors currently so let's move forward withn. Just use interactors to move forward.

jbolda committed:
add typescript as a dep within website for typedoc

jenweber committed:
Editing pass for the bigtest platform

jorgelainfiesta committed:
Spellcheck

Co-authored-by: Jen Weber <weberj10@gmail.com>

jorgelainfiesta committed:
Adjusts font sizes

jorgelainfiesta committed:
Changes baseURL to /bigtest

jorgelainfiesta committed:
Move /website/docs to /website

jorgelainfiesta committed:
Updates .gitignore

jorgelainfiesta committed:
Sets deployment to a subdirectory

jorgelainfiesta committed:
Moves .netlify ignore rule to root

jorgelainfiesta committed:
Untracks ignored file

jorgelainfiesta committed:
Update CSS url to include baseurl

jorgelainfiesta committed:
Adds ascii player via iframe to platform landing page

jorgelainfiesta committed:
Adds architecture info and quote

minkimcello committed:
Remove node from prerequisites

jbolda committed:
bump to version that supports multiple instances of plugin

jbolda committed:
adjust config for folder shift

jorgelainfiesta committed:
Graceful initial state for the iframe

minkimcello committed:
Fix readme path for plugin

minkimcello committed:
Add todo comments from feedback

jorgelainfiesta committed:
Adds demo to interactors landing page

jorgelainfiesta committed:
Apply suggestions from code review

Use `exists` on examples

Co-authored-by: Charles Lowell <cowboyd@frontside.com>

jorgelainfiesta committed:
Adds space in jest example

minkimcello committed:
Big changes to built-in and locators-filters-actions pages

minkimcello committed:
Improvements in the write-your-own page

minkimcello committed:
Fix broken link

jorgelainfiesta committed:
Makes styles corrections for background image

jorgelainfiesta committed:
Changes codeblock theme to nightowl

minkimcello committed:
Sync with Jen

minkimcello committed:
Replace placeholder links with appropriate API links

jorgelainfiesta committed:
Updates about copy with edited version

minkimcello committed:
Reorganize the introduction of the sample app

minkimcello committed:
Include information on how interactors return an array

minkimcello committed:
Add HTML markup example for table

minkimcello committed:
Add section on parallel testing in platform docs

minkimcello committed:
Update notes

jorgelainfiesta committed:
Corrects wording on report results

Co-authored-by: Charles Lowell <cowboyd@frontside.com>

jorgelainfiesta committed:
Improves copy on size section

minkimcello committed:
Add more information on actions

minkimcello committed:
Add date picker example for bottom of quick start

minkimcello committed:
Test format

minkimcello committed:
Better delegation example/explanation

minkimcello committed:
Temporary heading

jorgelainfiesta committed:
Fixes anchor scroll

jorgelainfiesta committed:
Removes typedoc generated API from docusaurus

jorgelainfiesta committed:
Set all links to API to point to /

jorgelainfiesta committed:
Adds discussion prompt at the end of pages

Finishes warning comment

minkimcello committed:
Add disclaimer about createInteractor api

jenweber committed:
Polish the quick start and write your own

minkimcello committed:
Remove more broken links

minkimcello committed:
Remove api link from footer

minkimcello committed:
Add source code links and update text so it says source code and not api

minkimcello committed:
Update notes

jorgelainfiesta committed:
Adds admonition styles + small tweaks

minkimcello committed:
Reenable the workflows

cowboyd committed:
resolve merge conflicts in yarn.lock

jbolda committed:
use lock.file from v0 branch

cowboyd reviewed (approved):
Let's roll 🙌🏻
